test/unit/StakingPool/depositTo.js

```diff
@@ -0,0 +1,774 @@
+
+    const newStakeShares = await estimateStakeShares({ amount, stakingPool });
+
+    // deposit to the same tokenId
+    await stakingPool.connect(user).depositTo([
+      {
+        amount,
+        trancheId: firstActiveTrancheId,
+        tokenId: depositNftId,
+        destination,
+      },
+    ]);
+
+    const updatedDepositData = await stakingPool.deposits(depositNftId, firstActiveTrancheId);
+    const newRewardShares = await getNewRewardShares({
+      stakingPool,
+      initialStakeShares: firstDepositData.stakeShares,
+      stakeSharesIncrease: newStakeShares,
+      initialTrancheId: firstActiveTrancheId,
+      newTrancheId: firstActiveTrancheId,
+    });
+    expect(updatedDepositData.pendingRewards).to.equal(0);
+    expect(updatedDepositData.lastAccNxmPerRewardShare).to.equal(0);
+    expect(updatedDepositData.stakeShares).to.equal(firstDepositData.stakeShares.add(newStakeShares));
+    expect(updatedDepositData.rewardsShares).to.equal(firstDepositData.rewardsShares.add(newRewardShares));
+  });
+
+  it('updates deposit pendingRewards and lastAccNxmPerRewardShare', async function () {
+    const { stakingPool } = this;
+    const {
+      members: [user],
+    } = this.accounts;
+
+    const { amount, tokenId, destination, depositNftId } = depositToFixture;
+
+    const { firstActiveTrancheId } = await getTranches();
+
+    // first deposit
+    await stakingPool.connect(user).depositTo([
+      {
+        amount,
+        trancheId: firstActiveTrancheId,
+        tokenId,
+        destination,
+      },
+    ]);
+
+    const depositData = await stakingPool.deposits(depositNftId, firstActiveTrancheId);
+
+    expect(depositData.lastAccNxmPerRewardShare).to.equal(0);
+    expect(depositData.pendingRewards).to.equal(0);
+
+    // Generate rewards
+    const coverRequest = {
+      coverId: 0,
+      productId: 0,
+      amount: parseEther('1000000'),
+      period: 3600 * 24 * 30, // 30 days
+      gracePeriod: 3600 * 24 * 30,
+      globalCapacityRatio: 20000,
+      capacityReductionRatio: 0,
+      rewardRatio: 10000,
+    };
+    await stakingPool.connect(this.coverSigner).allocateStake(coverRequest);
+
+    await increaseTime(
+      20 * // days
+        24 * // hours
+        60 * // minutes
+        60, // seconds
+    );
+
+    // Second deposit
+    await stakingPool.connect(user).depositTo([
+      {
+        amount,
+        trancheId: firstActiveTrancheId,
+        tokenId: depositNftId,
+        destination,
+      },
+    ]);
+
+    const secondDepositData = await stakingPool.deposits(depositNftId, firstActiveTrancheId);
+    const secondAccNxmPerRewardsShare = await stakingPool.accNxmPerRewardsShare();
+
+    expect(secondAccNxmPerRewardsShare).to.not.equal(0);
+    expect(secondDepositData.lastAccNxmPerRewardShare).to.equal(secondAccNxmPerRewardsShare);
+    // TODO: Shouldn't pendingRewards also be updated?
+    expect(secondDepositData.pendingRewards).to.equal(0);
```

Comment:
`deposit.pendingRewards` is not being increased in the second deposit because `deposit.lastAccNxmPerRewardShare` is set at the end of the second deposit, so it is only updated on the third deposit.

Is this the expected behavior? Based on the comment, it looks like the intention is to update `pendingRewards` after the first deposit. Should we use a different variable such as `deposit.rewardsShares != 0` on this `if` statement?  
  
https://github.com/NexusMutual/smart-contracts/blob/896dbdcb08cc4a89191ef133087d1e9d72ac6517/contracts/modules/staking/StakingPool.sol#L408-L412

Reply:
Using `deposit.rewardShares`  seems like a good fix to me. Can you confirm this is the expected behavior @shark0der?

Reply:
@shark0der Applied the fix here https://github.com/NexusMutual/smart-contracts/pull/495/commits/2e47b2f0d9285d5a1878ec847276a44bbc4fe9ed